storage/innobase/xtrabackup/doc/source/advanced/locks.rst

```diff
@@ -0,0 +1,79 @@
+.. _lock_redesign:
+
+===========================================
+``lock-ddl-per-table`` Option Improvements
+===========================================
+
+|MySQL| 5.7 has made improvements to bulk loading, one of these improvements
+is the ability to disable redo logging, which increased the speed of
+ADD INDEX operations.
+
+To block DDL statements on an instance, Percona Server implemented
+LOCK TABLES FOR BACKUP. |Percona XtraBackup| uses this lock for the duration
+of the backup. This lock does not affect DML statements.
+
+|Percona XtraBackup| has also implemented ``--lock-ddl-per-table``, which
+blocks DDL statements by using metadata locks (MDL).
+
+The following procedures describe a simplified backup operation when using
+``--lock-ddl-per-table``:
+
+1. Parse and copy all redo logs after the checkpoint mark
+
+2. Fork a dedicated thread to continue following new redo log entries
+
+3. List the tablespaces required to copy
+
+4. Iterate through the list. The following steps occur with each listed tablespace:
+
+    a. Query INFORMATION_SCHEMA.INNODB_SYS_TABLES to find which tables belong
+    to the tablespace ID and take a MDL on the underlying table or tables
+    in case there is a shared tablespace.
+    
+    b. Copy the tablespace ``.ibd`` files.
+    
+The backup process may encounter a redo log event, generated by the bulk load
+operations, which notifies backup tools that data file changes have been
+omitted from the redo log. This event is an ``MLOG_INDEX_LOAD``. If this
+event is found by the redo follow thread, the backup continues and assumes
+the backup is safe because the MDL protects tablespaces already copied and
+the MLOG_INDEX_LOAD event is for a tablespace that is not copied.
+
+These assumptions may not be correct and may lead to inconsistent backups.
+
+``--lock-ddl-per-table`` redesign
```

Comment:
Should I remove this section since --lock-ddl-per-table is deprecated?

Reply:
No, the same redesign applies to 8.0, but we are encouraging users to use --lock-ddl instead. On 2.4 lock-ddl was only available on PS and on 8.0 it is available in upstream and PS.